src/agent/src/main.rs

```diff
@@ -375,6 +402,110 @@ async fn start_sandbox(
     Ok(())
 }
 
+// If we fail to start the AA, ocicrypt won't be able to unwrap keys
+// and container decryption will fail.
+fn init_attestation_agent(logger: &Logger, _config: &AgentConfig) -> Result<()> {
+    let config_path = OCICRYPT_CONFIG_PATH;
+
+    // The image will need to be encrypted using a keyprovider
+    // that has the same name (at least according to the config).
+    let ocicrypt_config = serde_json::json!({
+        "key-providers": {
+            "attestation-agent":{
+                "ttrpc":AA_KEYPROVIDER_URI
+            }
+        }
+    });
```

Comment:
Now the KeyProvider API is moved from AA to [CDH](https://github.com/confidential-containers/guest-components/blob/main/confidential-data-hub/hub/src/bin/confidential-data-hub/server/mod.rs#L103-L156) and the socket file is `unix:///run/confidential-containers/cdh.sock`. 

So the UnwrapKey API should be depended on CDH. 

This means that it might be better to move the `ocicrypt_config` things to the function that launches CDH, and looks like
```json
{
    "key-providers": {
        "attestation-agent": {
            "ttrpc": "unix:///run/confidential-containers/cdh.sock"
        }
    }
}
```

[This](https://github.com/confidential-containers/guest-components/blob/main/image-rs/test_data/ocicrypt_keyprovider_ttrpc.conf) is test in [the integration test of image-rs](https://github.com/confidential-containers/guest-components/blob/main/image-rs/tests/image_decryption.rs#L59)

Also if `aa_kbc_params` is set to `cc_kbc::*`, AA is also required. If it is set to `offline_fs_kbc::*` or `online_sev_kbc::*`, we do not need AA.

Reply:
fixed, thanks